src/main/java/frc/robot/RobotState.java

```diff
@@ -102,6 +102,12 @@ public Pose2d getGlobalPoseEstimate() {
     private void addVisionObservation(Pose3d cameraPose, Pose3d robotPose, double timestamp,
         Vector<N3> baseUncertainty, List<PhotonTrackedTarget> targets, String prefix,
         boolean doInit, Circle circle) {
+        if (robotPose.getX() < 0 || robotPose.getY() < 0
+            || robotPose.getX() > FieldConstants.fieldLength.in(Meters)
+            || robotPose.getY() > FieldConstants.fieldWidth.in(Meters)
+                && Constants.StateEstimator.keepInField) {
```

Comment:
This will not do what u think it will. Look at https://docs.oracle.com/javase/tutorial/java/nutsandbolts/operators.html and see whats wrong.